src/ID3Util.ts

```diff
@@ -11,34 +11,25 @@ export class SplitBuffer {
     }
 }
 
+/**
+ * Expects a buffer containing a string at the beginning that is terminated by a \0 character.
+ * Returns a split buffer containing the bytes before and after null termination.
+ */
 export function splitNullTerminatedBuffer(buffer: Buffer, encodingByte = 0x00) {
-    const termination = { start: -1, size: 0 }
-    if (encodingByte === 0x01 || encodingByte === 0x02) {
-        termination.start = buffer.indexOf(Buffer.from([0x00, 0x00]))
-        termination.size = 2
-        if (
-            termination.start !== -1 &&
-            buffer.length > (termination.start + termination.size)
-        ) {
-            if (buffer[termination.start + termination.size] === 0x00) {
-                termination.start += 1
-            }
+    // UTF-16/BE always uses two bytes per character.
+    // \0 is therefore encoded as [0x00, 0x00] instead of just [0x00].
+    // We'll do a sliding window search, window size depends on encoding.
```

Comment:
I understand that you document it, as your first understanding of the how UTF16 was encoded was incorrect. However, this is pretty well known fact, I am not sure we need to document.
The new algorithm is way better and quite explicit, I don't think we need to explain it here.
Additionally if we refer to the `TextEncoding` definition, this also explained there.